sql/core/src/main/scala/org/apache/spark/sql/execution/basicOperators.scala

```diff
@@ -65,6 +65,21 @@ case class Filter(condition: Expression, child: SparkPlan) extends UnaryNode {
  * :: DeveloperApi ::
  */
 @DeveloperApi
+case class CustomFilter(condition: Expression, child: SparkPlan, out: Seq[Attribute]) 
+  extends UnaryNode {
+  override def output: Seq[Attribute] = out
```

Comment:
This is not correct.  You are claiming to produce a given output schema, but really the output of this operator will always be `child.output`.